ksql-common/src/main/java/io/confluent/ksql/util/SchemaUtil.java

```diff
@@ -348,24 +349,57 @@ public static boolean areCompatible(final Schema arg1, final Schema arg2) {
   }
 
   private static boolean mapEquals(final Schema mapA, final Schema mapB) {
-    return Objects.equals(mapA.keySchema(), mapB.keySchema())
-        && Objects.equals(mapA.valueSchema(), mapB.valueSchema());
+    return areCompatible(mapA.keySchema(), mapB.keySchema())
```

Comment:
given the change in semantics from `equals` to `compatible` I wonder if we should change the name of these methods? eg. `mapCompatible`